naptime/src/main/scala/org/coursera/naptime/actions/RestActionBuilder.scala

```diff
@@ -67,16 +85,42 @@ class RestActionBuilder[RACType, AuthType, BodyType, ResourceKeyType, ResourceTy
     new RestActionBuilder(auth, bodyParser, errorHandler)
 
   /**
-   * Set the body type.
+   * Like [[auth]] above, but with a body-aware generator function.
+   *
+   * Note that this version of [[auth]] converts the builder into a
+   * [[DefinedBodyTypeRestActionBuilder]], which forbids request body definition changes.
+   */
+  def bodyAuth[NewAuthType](authGenerator: BodyType => HeaderAccessControl[NewAuthType])
```

Comment:
As suggested in our private repo by @mbarackman-coursera , it's may be best to only allow allow this from `DefinedBodyTypeRestActionBuilder`, as default body is not meaningful. Thoughts from other reviewers?

Reply:
That restriction makes sense to me.